common/src/main/java/dev/xhyrom/lighteco/common/messaging/LightEcoMessagingService.java

```diff
@@ -0,0 +1,145 @@
+package dev.xhyrom.lighteco.common.messaging;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import dev.xhyrom.lighteco.api.messenger.IncomingMessageConsumer;
+import dev.xhyrom.lighteco.api.messenger.Messenger;
+import dev.xhyrom.lighteco.api.messenger.MessengerProvider;
+import dev.xhyrom.lighteco.api.messenger.message.Message;
+import dev.xhyrom.lighteco.api.messenger.message.type.UserUpdateMessage;
+import dev.xhyrom.lighteco.common.cache.ExpiringSet;
+import dev.xhyrom.lighteco.common.messaging.message.MessageType;
+import dev.xhyrom.lighteco.common.messaging.message.UserUpdateMessageImpl;
+import dev.xhyrom.lighteco.common.model.currency.Currency;
+import dev.xhyrom.lighteco.common.model.user.User;
+import dev.xhyrom.lighteco.common.plugin.LightEcoPlugin;
+import dev.xhyrom.lighteco.common.util.gson.GsonProvider;
+import lombok.Getter;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+public class LightEcoMessagingService implements InternalMessagingService, IncomingMessageConsumer {
+    @Getter
+    private final LightEcoPlugin plugin;
+    private final ExpiringSet<UUID> receivedMessages;
+
+    private final Messenger messenger;
+    private final MessengerProvider provider;
+
+    public LightEcoMessagingService(LightEcoPlugin plugin, MessengerProvider provider) {
+        this.plugin = plugin;
+
+        this.provider = provider;
+        this.messenger = provider.obtain(this);
+        this.receivedMessages = new ExpiringSet<>(30, TimeUnit.MINUTES);
+    }
+
+    private UUID generateMessageId() {
+        UUID id = UUID.randomUUID();
+
+        this.receivedMessages.add(id);
+        return id;
+    }
+
+    @Override
+    public void pushUserUpdate(User user, Currency currency) {
+        this.plugin.getBootstrap().getScheduler().async().execute(() ->
+                this.messenger.sendOutgoingMessage(
```

Comment:
make this to accept server name, and listen to multiple channels -> global channel, server name channel